From: Sam James <sam@gentoo.org>
Date: Fri, 1 Nov 2024 18:34:52 +0300
Subject: media-libs/webrtc-audio-processing: fix x86 w/o SSE
Origin: https://gitweb.gentoo.org/repo/gentoo.git/tree/media-libs/webrtc-audio-processing/files/webrtc-audio-processing-1.3-x86-no-sse.patch?id=29cd0e622b574df6adff5704ab4e220709619767
Bug: https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing/-/issues/5
Bug-gentoo: https://bugs.gentoo.org/921140
Last-Update: 2024-11-01

Tested with CFLAGS="-O2" CFLAGS_x86="-m32 -O2 -mno-sse -march=i686 -fcf-protection=none" e webrtc-audio-processing-1.3-r3.ebuild clean compile.

Signed-off-by: Sam James <sam@gentoo.org>
---
 webrtc/rtc_base/system/arch.h        | 2 +-
 webrtc/third_party/pffft/meson.build | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/webrtc/rtc_base/system/arch.h b/webrtc/rtc_base/system/arch.h
index aee37561d1f7..21426039620f 100644
--- a/webrtc/rtc_base/system/arch.h
+++ b/webrtc/rtc_base/system/arch.h
@@ -34,7 +34,7 @@
 #else
 #define WEBRTC_ARCH_32_BITS
 #endif
-#elif defined(_M_IX86) || defined(__i386__)
+#elif defined(__SSE__) && (defined(_M_IX86) || defined(__i386__))
 #define WEBRTC_ARCH_X86_FAMILY
 #define WEBRTC_ARCH_X86
 #define WEBRTC_ARCH_32_BITS
diff --git a/webrtc/third_party/pffft/meson.build b/webrtc/third_party/pffft/meson.build
index c1eb5c61d178..7059840ca2f5 100644
--- a/webrtc/third_party/pffft/meson.build
+++ b/webrtc/third_party/pffft/meson.build
@@ -4,7 +4,7 @@ pffft_sources = [
 
 pffft_cflags = [ '-D_GNU_SOURCE' ]
 
-if (have_arm and not have_neon) or (have_mips and host_machine.endian() == 'little') or have_mips64
+if (host_machine.cpu_family() == 'x86') or (have_arm and not have_neon) or (have_mips and host_machine.endian() == 'little') or have_mips64
   pffft_cflags += [ '-DPFFFT_SIMD_DISABLE' ]
 endif
 
-- 
2.45.2

